pkg/asset/machines/master_test.go

```diff
@@ -10,16 +11,57 @@ import (
 	"github.com/openshift/installer/pkg/asset"
 	"github.com/openshift/installer/pkg/asset/ignition/machine"
 	"github.com/openshift/installer/pkg/asset/installconfig"
+	"github.com/openshift/installer/pkg/asset/machines/machineconfig"
 	"github.com/openshift/installer/pkg/asset/rhcos"
 	"github.com/openshift/installer/pkg/types"
 	awstypes "github.com/openshift/installer/pkg/types/aws"
+	"github.com/vincent-petithory/dataurl"
 )
 
+func expectedMachineConfigForWorkloadCpuset(role, cpuset string) string {
+	expectedCrioCfg, _ := machineconfig.CrioWorkloadDropinContents(string(types.ManagementWorkloadPartition), cpuset)
+	expectedKubeletCfg, _ := machineconfig.KubeletWorkloadDropinContents(string(types.ManagementWorkloadPartition), cpuset)
+	return fmt.Sprintf(`apiVersion: machineconfiguration.openshift.io/v1
+kind: MachineConfig
+metadata:
+  creationTimestamp: null
+  labels:
+    machineconfiguration.openshift.io/role: %[1]s
+  name: 02-%[1]s-workload-partitioning
+spec:
+  config:
+    ignition:
+      version: 3.2.0
+    storage:
+      files:
+      - contents:
+          source: %s
+        mode: 420
+        overwrite: true
+        path: /etc/crio/crio.conf.d/01-management-workload
+        user:
+          name: root
+      - contents:
+          source: %s
+        mode: 420
+        overwrite: true
+        path: /etc/kubernetes/workload-pinning
+        user:
+          name: root
+  extensions: null
+  fips: false
+  kernelArguments: null
+  kernelType: ""
+  osImageURL: ""
+`, role, dataurl.EncodeBytes([]byte(expectedCrioCfg)), dataurl.EncodeBytes([]byte(expectedKubeletCfg)))
+}
+
 func TestMasterGenerateMachineConfigs(t *testing.T) {
 	cases := []struct {
 		name                  string
 		key                   string
 		hyperthreading        types.HyperthreadingMode
+		workloadCpuset        string
```

Comment:
Change this to `[]types.WorkloadPartition` and insert it directly into the InstallConfig when creating it.